src/vos/evtree.c

```diff
@@ -2131,6 +2131,21 @@ evt_large_hole_insert(daos_handle_t toh, const struct evt_entry_in *entry)
 	return rc == 0 ? alt_rc : rc;
 }
 
+static inline void
+evt_check_sort_policy(struct evt_context *tcx)
+{
+	/** EVT_FEAT_SORT_DIST can get imbalanced and, under certain conditions, can cause a
+	 *  segfault when the depth is larger than EVT_TRACE_MAX.  Rather than let it get too
+	 *  deep, let's detect cases where it starts getting deep and switch policies.
+	 */
+	if (unlikely(tcx->tc_depth > 5 &&
+		     (tcx->tc_feats & EVT_FEATS_SUPPORTED) == EVT_FEAT_SORT_DIST)) {
+		tcx->tc_feats ^= EVT_FEAT_SORT_DIST | EVT_FEAT_SORT_DIST_EVEN;
+		tcx->tc_ops = evt_policies[2];
+		D_DEBUG(DB_IO, "evtree depth is greater than 10, switching sort policy\n");
```

Comment:
5 or 10? "greater than tcx->tc_depth" to avoid misguiding.

Reply:
Good point. I initially picked 10.